app/src/main/kotlin/com/mapzen/erasermap/model/MapzenLocationImpl.kt

```diff
@@ -0,0 +1,62 @@
+package com.mapzen.erasermap.model
+
+import android.location.Location
+import com.mapzen.android.lost.api.LocationRequest
+import com.mapzen.android.lost.api.LocationServices
+import com.mapzen.android.lost.api.LostApiClient
+import com.mapzen.erasermap.EraserMapApplication
+import com.mapzen.pelias.PeliasLocationProvider
+import javax.inject.Inject
+
+public class MapzenLocationImpl(val app: EraserMapApplication) : MapzenLocation {
+    private val LOCATION_UPDATE_INTERVAL_IN_MS: Long = 1000L
+    private val LOCATION_UPDATE_SMALLEST_DISPLACEMENT: Float = 0f
+
+    var locationClient: LostApiClient? = null
+        @Inject set
+
+    init {
+        app.component()?.inject(this)
+    }
+
+    override fun connect() {
+        locationClient?.connect()
+    }
+
+    override fun disconnect() {
+        locationClient?.disconnect()
+    }
+
+    override fun isConnected(): Boolean {
+        val isConnected = locationClient?.isConnected()
+        if (isConnected is Boolean) return isConnected else return false
+    }
+
+    override fun initLocationUpdates(callback: (location: Location) -> Unit) {
+        if (locationClient?.isConnected() == false) {
```

Comment:
Good point.

Unfortunately you can't put a `!` in front of `locationClient?.isConnected()` because it returns type `Boolean?`

However just calling `isConnected()` on `this` does work since it returns type `Boolean` (not nullable). I will push an update.